### PR TITLE
chore(integ-test-alpha): add log group retention days to waiter function

### PR DESCRIPTION
**Pull request by sugardon**
### Issue # (if applicable)

None
(related to #29277)

### Reason for this change

When call  `waitForAssertions()`  the assertion provider creates CloudWatch log groups, but set to never expire

### Description of changes

Added log retention days to assertion provider in `WaiterStateMachine`

### Description of how you validated changes

Added unit test

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/integ-tests-alpha/lib/assertions/waiter-state-machine.ts [hunk 1]
```diff
@@ -33,6 +34,13 @@ export interface WaiterStateMachineOptions {
    * @default 1 (no backoff)
    */
   readonly backoffRate?: number;
+
+  /**
+   * How long, in days, the log contents will be retained.
+   *
+   * @default - no retention days specified
+   */
+  readonly logRetention?: RetentionDays;
```
Comment (msambol) [inline packages/@aws-cdk/integ-tests-alpha/lib/assertions/waiter-state-machine.ts:43]: for the maintainers to decide but do we actually want to set a reasonable default here? maybe 3 days or 1 week.

File: packages/@aws-cdk/integ-tests-alpha/lib/assertions/waiter-state-machine.ts [hunk 2]
```diff
@@ -33,6 +34,13 @@ export interface WaiterStateMachineOptions {
    * @default 1 (no backoff)
    */
   readonly backoffRate?: number;
+
+  /**
+   * How long, in days, the log contents will be retained.
+   *
+   * @default - no retention days specified
```
Comment (msambol) [inline packages/@aws-cdk/integ-tests-alpha/lib/assertions/waiter-state-machine.ts:41]: nit: could you remove the hyphen to match the other defaults ?

Comment (msambol) [reply packages/@aws-cdk/integ-tests-alpha/lib/assertions/waiter-state-machine.ts:41]: 👍🏼👍🏼

**Comment by TheRealAmazonKendra** [comment]
So, here's what I'm wondering overall: instead of a person having to specify this in each and every location we create these logs, could we just use where this is set on the assertions and then add the same here? Or, could we set this one the IntegTest construct and then use that value everywhere within the test infrastructure?

File: packages/@aws-cdk/integ-tests-alpha/lib/assertions/waiter-state-machine.ts (hunk 1 above)
Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/integ-tests-alpha/lib/assertions/waiter-state-machine.ts:43]: I think that no default is correct here. It's what was also done here https://github.com/aws/aws-cdk/pull/29277#discussion_r1525419143.

**Comment by sugardon** [comment]
Right now, I think people should specify log retention every location, like below.

```ts
import * as logs from 'aws-cdk-lib/aws-logs';

declare const testCase: IntegTest;
declare const start: IApiCall;

const describe = testCase.assertions.awsApiCall('StepFunctions', 'describeExecution', {
  executionArn: start.getAttString('executionArn'),
  // already we can set log retention after #29277
  RetentionDays: logs.RetentionDays.ONE_WEEK,
}).expect(ExpectedResult.objectLike({
  status: 'SUCCEEDED',
})).waitForAssertions({
  totalTimeout: Duration.minutes(5),
  interval: Duration.seconds(15),
  backoffRate: 3,
  // this PR
  logRetention: logs.RetentionDays.ONE_WEEK,
});
```

> could we set this one the IntegTest construct and then use that value everywhere within the test infrastructure?

I think it's a good interface! But, I'm not sure how to implement it right now. If possible, I would like to implement it.

File: packages/@aws-cdk/integ-tests-alpha/README.md [hunk 3]
```diff
@@ -555,6 +557,7 @@ const describe = testCase.assertions.awsApiCall('StepFunctions', 'describeExecut
   totalTimeout: Duration.minutes(5),
   interval: Duration.seconds(15),
   backoffRate: 3,
+  logRetention: logs.RetentionDays.ONE_WEEK,
```
Comment (comcalvi) [inline packages/@aws-cdk/integ-tests-alpha/README.md:560]: could we add this property to an existing integration test?

**Review by comcalvi — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/integ-tests-alpha/README.md (hunk 3 above)
Comment (sugardon) [reply packages/@aws-cdk/integ-tests-alpha/README.md:560]: Thank you for instructive comment! 
I think, it might take time, but I will try it!

**Pull request closed without merging**